src/services/api/me/me.service.ts

```diff
@@ -142,40 +144,33 @@ export class MeService {
         );
         continue;
       }
-      if (!showOnlyMyCreatedSpaces) {
-        mySpaceResults.push({
-          space: activityLog.space,
-          latestActivity: activityLog,
-        });
-      } else {
-        if (activityLog?.space) {
-          if (!activityLog?.space.community) {
-            this.logger.warn(
-              `Unable to process space entry ${activityLog?.space.id} because it does not have a community.`,
-              LogContext.ACTIVITY
-            );
-            continue;
-          }
-          const myRoles = await this.communityRoleService.getCommunityRoles(
-            agentInfo,
-            activityLog.space.community
-          );
-          if (
-            myRoles.includes(CommunityRole.ADMIN) &&
-            activityLog.space.level === 0
-          ) {
-            mySpaceResults.push({
-              space: activityLog.space,
-              latestActivity: activityLog,
-            });
-          }
-        }
-      }
+      mySpaceResults.push({
+        space: activityLog.space,
+        latestActivity: activityLog,
+      });
     }
 
     return mySpaceResults.slice(0, limit);
   }
 
+  public async getMyCreatedSpaces(agentInfo: AgentInfo): Promise<ISpace[]> {
+    const user = await this.userService.getUserOrFail(agentInfo.userID);
+    if (!user) {
+      throw new EntityNotFoundException(
+        `User not found ${agentInfo.userID}`,
+        LogContext.COMMUNITY
+      );
+    }
+    const accounts =
```

Comment:
In case of no accounts hosted by the user, can we just make sure the server returns an empty array?

Reply:
yeah, accounts was already an empty array, but now `compact` will make sure of that